### PR TITLE
feat(stakingEventWatcher): implement ADR-26 pre-approval staking flow and inclusion proof reporting

### PR DESCRIPTION
**Pull request by Lazar955**
### Summary
This PR implements the necessary changes to support the [ADR-26](https://github.com/babylonlabs-io/pm/blob/main/adr/adr-026-pre-approval-staking-flow.md) pre-approval staking flow.

### Key Changes
- **Renaming:** The `unbondingWatcher` has been renamed to `stakingEventWatcher` to reflect its broader responsibilities.
- **Functionality Updates:**
  - Continues to report unbonding transactions as before.
  - Now reports inclusion proofs for staking transactions to Babylon.
- **E2E** Introduced a new e2e test that verifies a delegation created without an inclusion proof will become active once the proof is provided.

### Related Issue
Closes [#67](https://github.com/babylonlabs-io/vigilante/issues/67)

### CODE REVIEW

File: btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go [hunk 1]
```diff
@@ -0,0 +1,550 @@
+package stakingeventwatcher
```
Comment (Lazar955) [inline btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go:1]: not sure why git don't see this as rename

File: btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+			sew.metrics.NumberOfTrackedActiveDelegations.Dec()
+
+		case <-sew.quit:
+			sew.logger.Debug("handle delegations loop quit")
+			return
+		}
+	}
+}
+
+func (sew *StakingEventWatcher) handlerVerifiedDelegations() {
```
Comment (Lazar955) [inline btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go:457]: 🍖  and 🥔  of the PR

File: btcstaking-tracker/stakingeventwatcher/expected_babylon_client.go [hunk 3]
```diff
@@ -130,3 +160,27 @@ func (bca *BabylonClientAdapter) BtcClientTipHeight() (uint32, error) {
...
+	msg := btcstakingtypes.MsgAddBTCDelegationInclusionProof{
+		Signer:                  signer,
+		StakingTxHash:           stakingTxHash.String(),
+		StakingTxInclusionProof: btcstakingtypes.NewInclusionProofFromSpvProof(proof),
+	}
+
+	resp, err := bca.babylonClient.ReliablySendMsg(ctx, &msg, []*errors.Error{}, []*errors.Error{})
+
+	if err != nil && resp != nil {
+		return fmt.Errorf("msg MsgBTCUndelegate failed exeuction with code %d and error %w", resp.Code, err)
```
Comment (KonradStaniec) [inline btcstaking-tracker/stakingeventwatcher/expected_babylon_client.go:178]: msg `MsgAddBTCDelegationInclusionProof`  failed to execute here

File: btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+	}
+
+	utils.PushOrQuit[*delegationInactive](
+		sew.unbondingRemovalChan,
+		&delegationInactive{stakingTxHash: delegationId},
+		sew.quit,
+	)
+}
+
+func (sew *StakingEventWatcher) handleDelegations() {
```
Comment (KonradStaniec) [inline btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go:405]: maybe lets call it ` handleUnbondedDelegations` ?

File: btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+			del := &newDelegation{
+				stakingTxHash:         stakingTxHash,
+				stakingTx:             delegation.StakingTx,
+				stakingOutputIdx:      delegation.StakingOutputIdx,
+				delegationStartHeight: delegation.DelegationStartHeight,
+				unbondingOutput:       delegation.UnbondingOutput,
+			}
+
+			// if we already have this delegation, skip it
+			if sew.unbondingTracker.GetDelegation(stakingTxHash) == nil && delegation.HasProof {
```
Comment (KonradStaniec) [inline btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go:192]: hmm does this mean we won't be tracking unbodnign of delegations without inclusions proof ? 
If yes then we need to improve this, we need to be able to also report unbondings of staking transactions that do not have inclusion proof on babylon.

File: btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go [hunk 6]
```diff
@@ -0,0 +1,551 @@
...
+	unbondingSignature *schnorr.Signature,
+) {
+	_ = retry.Do(func() error {
+		active, err := sew.babylonNodeAdapter.IsDelegationActive(stakingTxHash)
+
+		if err != nil {
+			return fmt.Errorf("error checking if delegation is active: %v", err)
+		}
+
+		if !active {
```
Comment (KonradStaniec) [inline btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go:337]: we should report unbonding transaction to Babylon even if it not active i.e babylon will accept unbonding transactions report if delegation is:
- either active
- or verified
and we should stop reporting if it is unbonded

File: btcstaking-tracker/stakingeventwatcher/expected_babylon_client.go [hunk 7]
```diff
@@ -44,10 +44,29 @@ func NewBabylonClientAdapter(babylonClient *bbnclient.Client) *BabylonClientAdap
 	}
 }
 
-// TODO: Consider doing quick retries for failed queries.
-func (bca *BabylonClientAdapter) ActiveBtcDelegations(offset uint64, limit uint64) ([]Delegation, error) {
+// BtcDelegations - returns btc delegations TODO: Consider doing quick retries for failed queries.
+func (bca *BabylonClientAdapter) BtcDelegations(offset uint64, limit uint64) ([]Delegation, error) {
```
Comment (KonradStaniec) [inline btcstaking-tracker/stakingeventwatcher/expected_babylon_client.go:48]: I think this is now unused, so lets delete it

**Review by KonradStaniec — APPROVED** [review]
LGTM

**Pull request merged**